### ISSUE TITLE
New version on simtk does not run (OSX)

### ISSUE THREAD

**Issue by jmerkow** [SimVascular/SimVascular#198]
I just downloaded the new version of simvascular on simtk and it does not run for me.
I am using El Captian.

Opening the app immediately crashes.  Running from CLI shows:

> $ /Applications/SimVascular.app/Contents/MacOS/SimVascular
> BUNDLE: /Applications/SimVascular.app
> RESOURCES: /Applications/SimVascular.app/Contents/Resources
> argc 1
> 
> 
>  *** simvascularApp: initializeLibraryPaths! ***
> 
> 
> Warning:  SV_PLUGIN_PATH doesn't exist!
> 
> Adding to plugin search path (/Applications/SimVascular.app/Contents/Resources/Bin/plugins)
> Adding to plugin search path (/Applications/SimVascular.app/Contents/Resources/Bin/lib/plugins)
> Adding to plugin search path (/Applications/SimVascular.app/Contents/Resources/Bin/../../plugins)
> Adding to plugin search path (/Applications/SimVascular.app/Contents/Resources/Bin/plugins)
> Adding to plugin search path (/Applications/SimVascular.app/Contents/Resources/Bin/bin/plugins)
> Adding to plugin search path (/Applications/SimVascular.app/Contents/Resources/Bin/lib/plugins)
> Wed Apr 12 18:32:28 2017 Debug: org.commontk.eventadmin.LogLevel=2
> in startOrUpdate, /Users/adamupdegrove/Documents/Software/SimVascular/MyMaster/Externals/Build/Externals/build/mitk-2016.03/ep/src/CTK/Plugins/org.commontk.eventadmin/ctkEAConfiguration.cpp:215
> Wed Apr 12 18:32:28 2017 Debug: org.commontk.eventadmin.CacheSize=30
> in startOrUpdate, /Users/adamupdegrove/Documents/Software/SimVascular/MyMaster/Externals/Build/Externals/build/mitk-2016.03/ep/src/CTK/Plugins/org.commontk.eventadmin/ctkEAConfiguration.cpp:217
> Wed Apr 12 18:32:28 2017 Debug: org.commontk.eventadmin.ThreadPoolSize=20
> in startOrUpdate, /Users/adamupdegrove/Documents/Software/SimVascular/MyMaster/Externals/Build/Externals/build/mitk-2016.03/ep/src/CTK/Plugins/org.commontk.eventadmin/ctkEAConfiguration.cpp:219
> Wed Apr 12 18:32:28 2017 Debug: org.commontk.eventadmin.Timeout=5000
> in startOrUpdate, /Users/adamupdegrove/Documents/Software/SimVascular/MyMaster/Externals/Build/Externals/build/mitk-2016.03/ep/src/CTK/Plugins/org.commontk.eventadmin/ctkEAConfiguration.cpp:221
> Wed Apr 12 18:32:28 2017 Debug: org.commontk.eventadmin.RequireTopic=true
> in startOrUpdate, /Users/adamupdegrove/Documents/Software/SimVascular/MyMaster/Externals/Build/Externals/build/mitk-2016.03/ep/src/CTK/Plugins/org.commontk.eventadmin/ctkEAConfiguration.cpp:223
> Wed Apr 12 18:32:28 2017 Debug: org.commontk.eventadmin.LogLevel=2
> in startOrUpdate, /Users/adamupdegrove/Documents/Software/SimVascular/MyMaster/Externals/Build/Externals/build/mitk-2016.03/ep/src/CTK/Plugins/org.commontk.eventadmin/ctkEAConfiguration.cpp:215
> Wed Apr 12 18:32:28 2017 Debug: org.commontk.eventadmin.CacheSize=30
> in startOrUpdate, /Users/adamupdegrove/Documents/Software/SimVascular/MyMaster/Externals/Build/Externals/build/mitk-2016.03/ep/src/CTK/Plugins/org.commontk.eventadmin/ctkEAConfiguration.cpp:217
> Wed Apr 12 18:32:28 2017 Debug: org.commontk.eventadmin.ThreadPoolSize=20
> in startOrUpdate, /Users/adamupdegrove/Documents/Software/SimVascular/MyMaster/Externals/Build/Externals/build/mitk-2016.03/ep/src/CTK/Plugins/org.commontk.eventadmin/ctkEAConfiguration.cpp:219
> Wed Apr 12 18:32:28 2017 Debug: org.commontk.eventadmin.Timeout=5000
> in startOrUpdate, /Users/adamupdegrove/Documents/Software/SimVascular/MyMaster/Externals/Build/Externals/build/mitk-2016.03/ep/src/CTK/Plugins/org.commontk.eventadmin/ctkEAConfiguration.cpp:221
> Wed Apr 12 18:32:28 2017 Debug: org.commontk.eventadmin.RequireTopic=true
> in startOrUpdate, /Users/adamupdegrove/Documents/Software/SimVascular/MyMaster/Externals/Build/Externals/build/mitk-2016.03/ep/src/CTK/Plugins/org.commontk.eventadmin/ctkEAConfiguration.cpp:223
> [Wed Apr 12 18:32:28 2017
> 1.623] Logfile: /Users/jameson/Library/Application Support/SimVascular/SimVascularApplication_2229601287/data/9/mitk-0.log
> /Applications/SimVascular.app/Contents/Resources/simvascular: line 73: 54984 Abort trap: 6           $SV_HOME/Bin/simvascular $*

Logfile contents:
> INFO|Wed Apr 12 18:32:28 2017 ||/Users/adamupdegrove/Documents/Software/SimVascular/MyMaster/Externals/Build/Externals/src/mitk-2016.03/Modules/Core/src/IO/mitkLog.cpp(130)|SetLogFile|0|MitkCore|Logfile: /Users/jameson/Library/Application Support/SimVascular/SimVascularApplication_2229601287/data/9/mitk-0.log

None of this is useful to me.  @updega2 perhaps you can make heads or tails of it?

**Comment by updega2** [comment]
Interesting. Should work fine on El Capitan. Can you see if the old GUI runs? Touch a file ~/.simvascular_default_tcl. And then run from command line.

**Comment by jmerkow** [comment]
Yes, that runs.  Obviously, it loads the tcl interface though.

**Comment by updega2** [comment]
Ya, just wanted to make sure it was an issue with loading the new GUI.

**Comment by updega2** [comment]
If possible, do you think you could remove .simvascular_default_tcl, export qt debug, and rerun and post the output.

export QT_DEBUG_PLUGINS=1

**Comment by jmerkow** [comment]
[simvascular_log-apr-13-2017.txt](https://github.com/SimVascular/SimVascular/files/919984/simvascular_log-apr-13-2017.txt)

**Comment by updega2** [comment]
Thanks! I'll take a look.

**Comment by updega2** [comment]
Weird, everything looks fine until it just dies. Can you try running in debug and see if that tells any more info? Add lldb in front of line 89 in /Application/Simvascular.app/Contents/Resources/simvascular

**Comment by jmerkow** [comment]
looks like your packages might be missing six

> BUNDLE: /Applications/SimVascular.app
> RESOURCES: /Applications/SimVascular.app/Contents/Resources
> (lldb) target create "/Applications/SimVascular.app/Contents/Resources/Bin/simvascular"
> Traceback (most recent call last):
>   File "<string>", line 1, in <module>
>   File "/Applications/Xcode.app/Contents/SharedFrameworks/LLDB.framework/Resources/Python/lldb/__init__.py", line 98, in <module>
>     import six
> ImportError: No module named six
> Traceback (most recent call last):
>   File "<string>", line 1, in <module>
> NameError: name 'run_one_line' is not defined
> Traceback (most recent call last):
>   File "<string>", line 1, in <module>
> NameError: name 'run_one_line' is not defined
> Traceback (most recent call last):
>   File "<string>", line 1, in <module>
> NameError: name 'run_one_line' is not defined

**Comment by updega2** [comment]
Can you still run in debug or does lldb not create a target because of that error?

**Comment by jmerkow** [comment]
> Process 65993 launched: '/Applications/SimVascular.app/Contents/Resources/Bin/simvascular' (x86_64)
> dyld: Library not loaded: lib_simvascular_post.dylib
>   Referenced from: /Applications/SimVascular.app/Contents/Resources/Bin/simvascular
>   Reason: image not found
> Process 65993 stopped
> * thread #1: tid = 0x16961b8, 0x00007fff5fc01075 dyld`dyld_fatal_error + 1, stop reason = EXC_BREAKPOINT (code=EXC_I386_BPT, subcode=0x0)
>     frame #0: 0x00007fff5fc01075 dyld`dyld_fatal_error + 1
> dyld`dyld_fatal_error:
> ->  0x7fff5fc01075 <+1>: nop
> 
> dyld`dyldbootstrap::start:
>     0x7fff5fc01076 <+0>: pushq  %rbp
>     0x7fff5fc01077 <+1>: movq   %rsp, %rbp
>     0x7fff5fc0107a <+4>: pushq  %r15

**Comment by jmerkow** [comment]
> ll /Applications/SimVascular.app/Contents/Resources/Lib
> total 56200
> -rwxr-xr-x@  1 jameson  admin    282680 Apr  8 20:47 lib_simvascular_adaptor.dylib
> -rwxr-xr-x@  1 jameson  admin    668644 Apr  8 20:47 lib_simvascular_geom.dylib
> -rwxr-xr-x@  1 jameson  admin      9336 Apr  8 20:47 lib_simvascular_globals.dylib
> -rwxr-xr-x@  1 jameson  admin    173716 Apr  8 20:47 lib_simvascular_image.dylib
> -rwxr-xr-x@  1 jameson  admin   3088812 Apr  8 20:47 lib_simvascular_itk_lset.dylib
> -rwxr-xr-x@  1 jameson  admin    401488 Apr  8 20:47 lib_simvascular_lset.dylib
> -rwxr-xr-x@  1 jameson  admin    200976 Apr  8 20:47 lib_simvascular_mesh.dylib
> -rwxr-xr-x@  1 jameson  admin    377740 Apr  8 20:47 lib_simvascular_mmg_mesh.dylib
> -rwxr-xr-x@  1 jameson  admin    301796 Apr  8 20:47 lib_simvascular_module_common.dylib
> -rwxr-xr-x@  1 jameson  admin    597128 Apr  8 20:47 lib_simvascular_module_mesh.dylib
> -rwxr-xr-x@  1 jameson  admin   1024348 Apr  8 20:47 lib_simvascular_module_model.dylib
> -rwxr-xr-x@  1 jameson  admin    596384 Apr  8 20:47 lib_simvascular_module_path.dylib
> -rwxr-xr-x@  1 jameson  admin    463568 Apr  8 20:47 lib_simvascular_module_projectmanagement.dylib
> -rwxr-xr-x@  1 jameson  admin    280768 Apr  8 20:47 lib_simvascular_module_qtwidgets.dylib
> -rwxr-xr-x@  1 jameson  admin    762780 Apr  8 20:47 lib_simvascular_module_segmentation.dylib
> -rwxr-xr-x@  1 jameson  admin    458552 Apr  8 20:47 lib_simvascular_module_simulation.dylib
> -rwxr-xr-x@  1 jameson  admin    522616 Apr  8 20:47 lib_simvascular_opencascade_solid.dylib
> -rwxr-xr-x@  1 jameson  admin    354200 Apr  8 20:47 lib_simvascular_polydata_solid.dylib
> -rwxr-xr-x@  1 jameson  admin    231092 Apr  8 20:47 lib_simvascular_post.dylib
> -rwxr-xr-x@  1 jameson  admin    100860 Apr  8 20:47 lib_simvascular_python_interp.dylib
> -rwxr-xr-x@  1 jameson  admin    144996 Apr  8 20:47 lib_simvascular_repository.dylib
> -rwxr-xr-x@  1 jameson  admin    157324 Apr  8 20:47 lib_simvascular_solid.dylib
> -rw-r--r--@  1 jameson  admin    148240 Apr  8 20:47 lib_simvascular_solverio.a
> -rwxr-xr-x@  1 jameson  admin    145340 Apr  8 20:47 lib_simvascular_tetgen_adaptor.dylib
> -rwxr-xr-x@  1 jameson  admin   1159608 Apr  8 20:47 lib_simvascular_tetgen_mesh.dylib
> -rw-r--r--@  1 jameson  admin   2697240 Apr  8 20:47 lib_simvascular_thirdparty_tetgen.a
> -rw-r--r--@  1 jameson  admin  10997000 Apr  8 20:47 lib_simvascular_thirdparty_vmtk.a
> -rw-r--r--@  1 jameson  admin    214472 Apr  8 20:47 lib_simvascular_thirdparty_zlib.a
> -rwxr-xr-x@  1 jameson  admin    184052 Apr  8 20:47 lib_simvascular_utils.dylib
> -rwxr-xr-x@  1 jameson  admin    291404 Apr  8 20:47 lib_simvascular_vtksvboolean.dylib
> -rwxr-xr-x@  1 jameson  admin    190184 Apr  8 20:47 lib_simvascular_vtksvcommon.dylib
> -rwxr-xr-x@  1 jameson  admin    435908 Apr  8 20:47 lib_simvascular_vtksvfilters.dylib
> -rwxr-xr-x@  1 jameson  admin    311336 Apr  8 20:47 lib_simvascular_vtksvgeometry.dylib
> -rwxr-xr-x@  1 jameson  admin    325424 Apr  8 20:47 lib_simvascular_vtksvnurbs.dylib
> -rwxr-xr-x@  1 jameson  admin    388372 Apr  8 20:47 lib_simvascular_vtksvparameterization.dylib
> drwxr-xr-x@ 10 jameson  admin       340 Apr  8 20:47 plugins

**Comment by updega2** [comment]
Interesting, but lib_simvascular_post is there.

**Comment by jmerkow** [comment]
Checking $SV_HOME....

**Comment by updega2** [comment]
export DYLD_PRINT_LIBRARIES=1 might help

**Comment by jmerkow** [comment]
Where do you want me to put that?

**Comment by updega2** [comment]
Put what? The export?

**Comment by jmerkow** [comment]
Ya

**Comment by updega2** [comment]
just export in terminal

**Comment by jmerkow** [comment]
It doesn't add any output with running simvascular on the CLI, as it usually does.

**Comment by updega2** [comment]
hmm, okay, it did for me. $SV_HOME was okay?

**Comment by updega2** [comment]
Probably because it tries to load lib_simvascular_post right away, you just get the error that it wasn't loaded.

**Comment by jmerkow** [comment]
I ran the entire startup script directly in the terminal and got this output.
[simlog2.txt](https://github.com/SimVascular/SimVascular/files/920120/simlog2.txt)

**Comment by jmerkow** [comment]
otool output: 
[simvasc_otool.txt](https://github.com/SimVascular/SimVascular/files/920124/simvasc_otool.txt)

**Comment by jmerkow** [comment]
Do you have a static library version I could try?

**Comment by jmerkow** [comment]
Could this be it? lib_simvascular_post.dylib has this in its otool:

> /Applications/SimVascular.app/Contents/Resources/Lib/lib_simvascular_post.dylib:
> 	lib_simvascular_post.dylib (compatibility version 0.0.0, current version 0.0.0)
> 	@rpath/libpython2.7.dylib (compatibility version 0.0.0, current version 0.0.0)
> 	/Users/adamupdegrove/Documents/Software/SimVascular/MyMaster/Externals/Build/Externals/bin/clang-7.0/x64/tcltk-8.6.4/lib/libtcl8.6.dylib (compatibility version 8.6.0, current version 8.6.4)
> 	/Users/adamupdegrove/Documents/Software/SimVascular/MyMaster/Externals/Build/Externals/bin/clang-7.0/x64/tcltk-8.6.4/lib/libtk8.6.dylib (compatibility version 8.6.0, current version 8.6.4)

I don't have these libs:
/Users/adamupdegrove/Documents/Software/SimVascular/MyMaster/Externals/Build/Externals/bin/clang-7.0/x64/tcltk-8.6.4/lib/libtcl8.6.dylib

**Comment by jmerkow** [comment]
Still why would `$SV_HOME/Bin/simvascular $SV_HOME/Tcl/SimVascular_2.0/simvascular_startup.tcl -tcl $*` but not $SV_HOME/Bin/simvascular.

**Comment by updega2** [comment]
I was thinking the same thing. That doesn't make sense.

**Comment by updega2** [comment]
You should have libtcl8.6.dylib under sv_externals/lib

**Comment by updega2** [comment]
No, I wish we had a static version, but with MITK plugins and modules, we need dynamic libs.

**Comment by updega2** [comment]
Wait, in simlog2.txt it shows the lib_simvascular_post.dylib was loaded.

**Comment by jmerkow** [comment]
Its odd.  Do you have a clean OS X you could test this on?

**Comment by updega2** [comment]
Yea, but only Yosemite. Thats what I test on.

**Comment by jmerkow** [comment]
Where do we go from here?

**Comment by updega2** [comment]
Not really sure. It seems kind of weird that you've been the only person to report an issue on start up. No offense :) Is there anyway there is something left over on your system for the good ol SimVascular days that could be mucking with things?

**Comment by updega2** [comment]
Since I have Yosemite, I'm going to see if I can download El Capitan, but not update, create an image, and put it in a VM alongside my Yosemite VM. I'll let you know if that works out.

**Comment by updega2** [comment]
Alright, I was able to set up an El Capitan VM, and SimVascular launches fine. Do you have anything set in PATH or DYLD_LIBRARY_PATH that could be throwing it off?

**Comment by jmerkow** [comment]
Nope. DYLD_LIBRARY_PATH is empty PATH just has normal stuff.

This is a new machine, it doesn't even have macports on it. Not sure what it could be.  Do I need to install Qt myself?

**Comment by updega2** [comment]
Nope, everything is in the installer. I didn't do anything but download, install, and launch on the clean El Capitan OS I installed in the VM this morning. That is really weird. Do have anymore than this in your path?:

/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin

**Comment by jmerkow** [comment]
Ok. I am going to try building it myself.  Does the super build still work? and do I need special compilers still?

**Comment by updega2** [comment]
No special compilers, clang works. But you obviously need Xcode with command line tools. You don't need macports. No more superbuild. MITK brings in a load of new dependencies, so I separated it out to the build of externals and the build of SimVascular. We provide pre-built binaries off the externals on the simvascular server, but they can be automatically download through SimVascular cmake for externals. See the documentation:

http://simvascular.github.io/docsCompile.html

Also, I just added mac to travis, and I believe that uses mac os x 10.11, so if you want to see how thats doing it, take a look at my branch updega2/2017-04-10-minor_updates
